core/dictgen/src/rootcling_impl.cxx

```diff
@@ -4506,14 +4506,8 @@ int RootClingMain(int argc,
       // rootcling
       if (interp.declare("namespace std {} using namespace std;") != cling::Interpreter::kSuccess
             // CINT uses to define a few header implicitly, we need to do it explicitly.
-            || interp.declare("#include <assert.h>\n"
-                              "#include <stdlib.h>\n"
-                              "#include <stddef.h>\n"
-                              "#include <string.h>\n"
-                             ) != cling::Interpreter::kSuccess
             || interp.declare("#include \"Rtypes.h\"\n"
-                              "#include \"TClingRuntime.h\"\n"
```

Comment:
We can kill `TClingRuntime.h` as this is its only use.

Reply:
Was the "why remove the other includes ... ? We should not rely on transitive inclusions through TObject.h."
resolved?

Reply:
I believe so - these were ROOT headers, and they are now not part of the change anymore.

Reply:
I am confused now ... I  thought this was talking about:
``` 
                            "#include <stdlib.h>\n"
                              "#include <stddef.h>\n"
                              "#include <string.h>\n"
```
which were the only header in TClingRunTime.h (+ stdio.h and math.h)

Reply:
They were targeted at an earlier version of this PR that removed `TObject.h` and `Rtypes.h`. See also my comment below

> rootcling needs to be exposed to types as before, in case people (maybe unknowingly) rely on e.g. <string> being included and using namespace std being injected (#pragma link C++ class MyClass<string>). OTOH, string.h is likely not needed in rootcling's runtime.